exonum-java-binding-qa-service/src/test/java/com/exonum/binding/qaservice/QaServiceImplIntegrationTest.java

```diff
@@ -308,6 +309,19 @@ void getBlockTransactions() {
     }
   }
 
+  @Test
+  void getActualConfigurationBeforeInit() {
+    assertThrows(IllegalStateException.class,
+        () -> service.getActualConfiguration());
+  }
+
+  @Test
+  void getActualConfiguration() {
+    setServiceNode(node);
+
+    assertNull(service.getActualConfiguration());
```

Comment:
Why must it be `null`? This doesn't seem correct.

Reply:
Tests in this class use either a mock of a `Node`, the interface that gives a read-only access to the database, if that access is not needed; and a fake, particularly, `NodeFake`. The fake allows to create snapshots from a memory database which can be populated with some data.